### ISSUE TITLE
WCF - Add methods to get entities in the sector

### ISSUE THREAD

**Issue by dodexahedron** [SEServerExtender/SEServerExtender#2]
The WCF service should expose more interesting data, such as CubeGrids, Characters, and FloatingObjects in the sector.

**Comment by zrisher** [comment]
+1, character and floatingobject cleanup is not great in Core right now, so this would be helpful.

**Comment by dodexahedron** [comment]
Can you elaborate a bit?  I'm not sure what you're asking for, there.

**Comment by zrisher** [comment]
I don't know how you use WCF, but I presume it is to communicate between Essentials and the Extender client. Thus I guessed that exposing these things would make managing them in the Extender client easier. Sorry if I misinterpreted!

**Comment by dodexahedron** [comment]
Plugins are loaded by reflection.
WCF is for external access, such as third party GUIs/management consoles/etc.

-----Original Message-----

**Comment by zrisher** [comment]
I.e. SESM

**Comment by dodexahedron** [comment]
Right. SESM is one of the intended consumers.  Scipio has a few open requests with me for that.
There's someone else who runs a couple servers and said he wants to make a dashboard, too, so there's definitely interest.
The main impetus, though, was to replace to broken external client for SESE and to ultimately de-couple the UI from SESE entirely.

**Comment by zrisher** [comment]
Yeah, SESE + SESM is a happy combo for me, and I know quite a few others. Would love to see tighter integration.

**Comment by devu** [comment]
I wish to see possibility to turn off WCF completely as I suspect it's a security kitchen doors. Did anyone without login into your server managed to paste 8500 block object killing server performance yet? It happened to me and it is my 1st suspect. So I am opposed to enhancing it any way as long as is running in background.

**Comment by dodexahedron** [comment]
That has absolutely nothing to do with WCF.

It’s actually a basic bug in the game and it’s fairly well-known among trolls.

WCF is authenticated, unless you manually configure it otherwise. So unless someone has stolen your administrator credentials anyway (as in the local windows administrator account), you’re safe.

**Comment by devu** [comment]
Thanks for correction here. I saw piece of code for cross server jump that using it and can push blocks from one server to another. That's why I draw that conclusion. Still though ability to disable it would be nice. I don't need extra hog on my instance. At current state every cycle is important.
